File: Homework/Ngan_Luu/78-subsets.py
#https://leetcode.com/problems/subsets/
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        n=len(nums)
        subset_list=[]
        for i in range(pow(2,len(nums))):
            subset=[]
            binary="{0:b}".format(i).zfill(len(nums))
            for i in range(len(binary)):
                if binary[i]=='1':
                    subset.append(nums[i])
            subset_list.append(subset)
        return subset_list

File: Homework/Son/W1/78-subsets.js
/**
 * @param {number[]} nums
 * @return {number[][]}
 */
 var subsets = function(nums) {
    const result = [];
  result.push([]);   
  
  let size = nums.length
  
  for(let i = 1; i < (1<<size) ; i++){  
    let subset = [];
    let bitmask=0
      while(bitmask<size){
          if(i & (1 << bitmask)){
              subset.push( nums[bitmask] );
              console.log(subset)
          }
          bitmask++   
      }
      result.push(subset)
  }
  return result
};

// https://leetcode.com/problems/subsets/